backends/redis.go

```diff
@@ -0,0 +1,118 @@
+package backends
+
+import (
+	"fmt"
+	"net/url"
+	"strconv"
+
+	"github.com/go-redis/redis"
+)
+
+type RedisBackend struct {
+	client *redis.Client
+}
+
+func NewRedisBackend(connString string) (*RedisBackend, error) {
+	u, _ := url.Parse(connString)
+	dbParam := u.Query().Get("database")
+
+	// Default to DB 0
+	var db int
+	if dbParam != "" {
+		fmt.Print(dbParam)
+		db, _ = strconv.Atoi(dbParam)
+	} else {
+		db = 0
+	}
+
+	host := u.Host
+	password := u.Query().Get("password")
+
+	client := redis.NewClient(&redis.Options{
+		Addr:     host,
+		Password: password,
+		DB:       db,
+	})
+
+	return &RedisBackend{
+		client: client,
+	}, nil
+}
+
+func (r *RedisBackend) Close() error {
+	return r.client.Close()
+}
+
+func (r *RedisBackend) CreateSchema() error {
+	return nil
+}
+
+func (r *RedisBackend) DropSchema() error {
+	r.client.FlushDb()
+	return nil
+}
+
+func (r *RedisBackend) CreateToken(token string) error {
+	err := r.client.Set(token, 0, 0).Err()
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (r *RedisBackend) IncrementAndGetNamespacedToken(token string, namespace string) (int64, error) {
+	nsToken := token + namespace
+	_, err := r.client.Get(nsToken).Result()
+
+	// If namespace doesn't exist, create it and set count to 0
+	if err != nil {
```

Comment:
I think there's a race condition here if two clients try and increment a namespaced token at the same time, right?

If I'm understanding the docs, `INCR` already handles this:

> Increments the number stored at key by one. If the key does not exist, it is set to 0 before performing the operation